pandas/tseries/index.py

```diff
@@ -827,16 +828,21 @@ def _format_native_types(self, na_rep=u('NaT'),
     def to_datetime(self, dayfirst=False):
         return self.copy()
 
-    def astype(self, dtype):
+    def astype(self, dtype, copy=True):
         dtype = np.dtype(dtype)
```

Comment:
Should I also add `copy` parameter to `Index` and modify the code there? And what about `Float64Index`? 
It may be tricky to figure out when a copy is made there and when not.